### PR TITLE
[WIP][DNM]Add periodicStart() to support cron builds

### PR DESCRIPTION
**Pull request by pranavrc**
--To be filled--

Do not merge.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -48,6 +51,66 @@ class ExecutorQueue extends Executor {
...
+     * @param {Object} config.job           Job object to create periodic builds for
+     * @param {Object} config.tokenGen      Function to generate JWT from username, scope and scmContext
+     * @return {Promise}
+     */
+    async _periodicStart(config) {
+        const pipeline = config.pipeline;
+        const job = config.job;
+        const tokenGen = config.tokenGen;
+
+        await this.scheduler.connect();
```
Comment (minzcmu) [inline index.js]: Do not create new connection everytime, it creates some weird effect. Try to do something like `this.connect()` function: https://github.com/screwdriver-cd/executor-queue/blob/608e644846536afb012f1eaba0acf8f71725c393/index.js#L171

**Review by minzcmu — CHANGES_REQUESTED** [review]

File: index.js [hunk 2]
```diff
@@ -48,6 +51,91 @@ class ExecutorQueue extends Executor {
...
+    }
+
+    /**
+     * Starts a new periodic build in an executor
+     * @method _startPeriodic
+     * @param {Object} config               Configuration
+     * @param {Object} config.pipeline      Pipeline of the job
+     * @param {Object} config.job           Job object to create periodic builds for
+     * @param {Object} config.tokenGen      Function to generate JWT from username, scope and scmContext
+     * @param {Boolean}config.update        Boolean to determine if updating existing periodic build
```
Comment (d2lam) [inline index.js]: should name this `isUpdate` to make it clearer

File: index.js [hunk 3]
```diff
@@ -48,6 +51,91 @@ class ExecutorQueue extends Executor {
...
+        this.scheduler = new Resque.Scheduler({ connection: redisConnection });
+    }
+
+    /**
+     * Starts a new periodic build in an executor
+     * @method _startPeriodic
+     * @param {Object} config               Configuration
+     * @param {Object} config.pipeline      Pipeline of the job
+     * @param {Object} config.job           Job object to create periodic builds for
+     * @param {Object} config.tokenGen      Function to generate JWT from username, scope and scmContext
```
Comment (d2lam) [inline index.js]: This is a Function

File: index.js [hunk 4]
```diff
@@ -48,6 +51,91 @@ class ExecutorQueue extends Executor {
...
+     * @param {Boolean}config.update        Boolean to determine if updating existing periodic build
+     * @return {Promise}
+     */
+    async _startPeriodic(config) {
```
Comment (d2lam) [inline index.js]: `async _startPeriodic({pipeline, job, tokenGen, update = false})`
and you can remove the next 4 lines for initialization.

File: index.js [hunk 5]
```diff
@@ -48,6 +51,91 @@ class ExecutorQueue extends Executor {
...
+        await this.connect();
+
+        const pipeline = config.pipeline;
+        const job = config.job;
+        const tokenGen = config.tokenGen;
+        const update = config.update || false;
+        const buildCron = job.annotations.buildPeriodically;
```
Comment (d2lam) [inline index.js]: this is all under `permutations`

**Review by d2lam — CHANGES_REQUESTED** [review]

**Comment by pranavrc** [comment]
Opened a new PR here: https://github.com/screwdriver-cd/executor-queue/pull/14

Closing this one.

**Pull request closed without merging**